core/services/workflows/testdata/fixtures/workflows/workflow_schema.json

```diff
@@ -1,14 +1,48 @@
 {
   "$schema": "https://json-schema.org/draft/2020-12/schema",
-  "$id": "https://github.com/smartcontractkit/chainlink/v2/core/services/workflows/workflow-spec",
-  "$ref": "#/$defs/workflowSpec",
+  "$id": "https://github.com/smartcontractkit/chainlink/v2/core/services/workflows/workflow-spec-yaml",
+  "$ref": "#/$defs/workflowSpecYaml",
   "$defs": {
-    "stepDefinition": {
-      "properties": {
-        "type": {
+    "stepDefinitionType": {
+      "oneOf": [
+        {
           "type": "string",
           "pattern": "^[a-z0-9_\\-:]+@(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$"
         },
+        {
+          "$schema": "https://json-schema.org/draft/2020-12/schema",
+          "$id": "https://github.com/smartcontractkit/chainlink/v2/core/services/workflows/step-definition-table-type",
+          "properties": {
+            "name": {
+              "type": "string"
```

Comment:
@HenryNguyen5 Shouldn't some of these properties be considered required? I think that's the case for the name and the version (although note we haven't been enforcing this anywhere atm and have often just used a name).

Reply:
They are required, see https://github.com/smartcontractkit/chainlink/pull/12622/files/114388ad37f0b5e7a2c7e17a4cc5602dbd057823#diff-8814ce758da6621323338140f96cee04e2a88737b03d64c447402025afe01641R32